File: AlgoExpert/Binary Trees/Easy/Node Depths.java
import java.util.*;

class Program {

  public static int nodeDepths(BinaryTree root) {
    // Write your code here.
    return nodeDepths(root, 0, 0);
  }
	
	public static int nodeDepths(BinaryTree root, int totalSum, int currentNodeDepth) {
    // Write your code here.
    if(root == null){
			return totalSum;
		}else{
			totalSum += currentNodeDepth;
			totalSum = nodeDepths(root.left, totalSum, currentNodeDepth + 1);
			totalSum = nodeDepths(root.right, totalSum, currentNodeDepth + 1);
		}
		
		return totalSum;
  }

  static class BinaryTree {
    int value;
    BinaryTree left;
    BinaryTree right;

    public BinaryTree(int value) {
      this.value = value;
      left = null;
      right = null;
    }
  }
}

File: Python/AlgoExpert/Binary Trees/Easy/nodeDepths.py
def nodeDepths(root, curr_depth=0):
    # Write your code here.
    if not root:
        return 0

    sum = 0
    sum += nodeDepths(root.left, curr_depth + 1)
    sum += nodeDepths(root.right, curr_depth + 1)
    sum += curr_depth

    return sum
    
    
# This is the class of the input binary tree.
class BinaryTree:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None
